app/helpers/custom_field_hierarchy_tree_view_helper.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module CustomFieldHierarchyTreeViewHelper
+  def render_tree(custom_field, tree_view, options = {})
+    hierarchy_hash = custom_field.hierarchy_root.hash_tree
+
+    return {} if hierarchy_hash.nil?
```

Comment:
:yellow_circle: Is the return value of this method relevant at all? Since it's always at the end of the `render` block for the tree view, it seems plausible. But on the other hand we'd return the result of `add_sub_tree` in the normal case, which would be just one random `each` result. To me it seems that the side effect of calling `add_sub_tree` is the important thing, because that calls methods on `tree_view`.

My current hypothesis is that the return value plays no role and we could simplify this to

```suggestion
    return if hierarchy_hash.nil?
```

Reply:
yeah, this is a copy paste error, good eye